packages/introspect/src/generate.ts

```diff
@@ -116,10 +148,25 @@ function createTableDefId(type: TableDetails, defaultSchema: string) {
     return `table_${type.schemaName}_${type.name}`;
 }
 
-function createTableJsonSchema(table: TableDetails): JSONSchema {
+function createTableJsonSchema(
+    table: TableDetails,
+    tableDefOverrides: TableDefOverrides | undefined,
+): JSONSchema {
     const empty: Record<string, any> = {};
 
+    // Here, we want to check if there was overriding table schema def
```

Comment:
I think you're missing an `applyOverrides` function that encapsulates this logic.